### PR TITLE
Frontend: New modal for Instance group deletion

### PR DESCRIPTION
**Pull request by cannarelladev**
With this PR will be implement and improve the system and the modal to delete a certain group of instances:

- [x] Active Page [Manager]: Selective group deletion
- [x] Active Page [Manager]: Group deletion by "Template"
- [x] Active Page [Personal]: Delete all personal instances (Persistent ones won't be deleted anymore)
- [x] Simple deletion modal restyling

### CODE REVIEW

File: frontend/src/utilsLogic.tsx [hunk 1]
```diff
@@ -384,27 +404,22 @@ export const getTemplatesMapped = (
...
-      id: ws.id,
-      title: ws.prettyName,
-      role: ws.role,
-      templates: templates.filter(({ workspaceId: id }) => id === ws.id),
-    }))
+    .map(ws =>
+      Object.assign({}, ws, {
+        templates: templates.filter(({ workspaceName: id }) => id === ws.name),
+      })
+    )
```
Comment (QcFe) [inline frontend/src/utilsLogic.tsx:422]:
```suggestion
    .map(ws => ({
      ...ws, 
      templates: templates.filter(({ workspaceName: id }) => id === ws.name),
    })
```

File: frontend/src/utilsLogic.tsx [hunk 2]
```diff
@@ -384,27 +404,22 @@ export const getTemplatesMapped = (
       persistent,
       resources: { cpu: 0, memory: '', disk: '' },
       instances: instancesSorted || instancesFiltered,
-      workspaceId,
+      workspaceName,
+      workspaceNamespace: 'workspace-' + workspaceName,
```
Comment (QcFe) [inline frontend/src/utilsLogic.tsx:408]: didn't we have workspaceNamespace attribute at a point?

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 3]
```diff
@@ -160,8 +159,12 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
       return i;
     });
 
+  //console.log(dataTemplate);
+
   const templates = joinInstancesAndTemplates(dataTemplate, dataInstances);
 
+  //console.log(templates);
+
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:167]:
```suggestion
```

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 4]
```diff
@@ -160,8 +159,12 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
       return i;
     });
 
+  //console.log(dataTemplate);
+
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:163]:
```suggestion
```

File: frontend/src/components/activePage/ActiveViewLogic/ActiveViewLogic.tsx [hunk 5]
```diff
@@ -12,17 +13,18 @@ const ActiveViewLogic: FC<{}> = ({ ...props }) => {
...
+    const { workspaceWrapperTenantV1alpha2, name, role } = workspace!;
+    const { spec, status } =
+      workspaceWrapperTenantV1alpha2?.itPolitoCrownlabsV1alpha1Workspace!;
+    return {
+      prettyName: spec?.prettyName as string,
+      role: WorkspaceRole[role!],
+      namespace: status?.namespace?.name!,
+      id: name!,
+    };
+  }) || []; */
```
Comment (QcFe) [inline frontend/src/components/activePage/ActiveViewLogic/ActiveViewLogic.tsx:27]:
```suggestion
```

**Review by QcFe — COMMENTED** [review]
some random suggestions

File: frontend/src/utilsLogic.tsx (hunk 2 above)
Comment (cannarelladev) [reply frontend/src/utilsLogic.tsx:408]: i don't believe at this point, here objects have been elaborated many times

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx [hunk 6]
```diff
@@ -105,6 +105,7 @@ const TableInstance: FC<ITableInstanceProps> = ({ ...props }) => {
             showHeader={false}
             pagination={false}
             rowClassName=""
+            rowKey={i => 1}
```
Comment (QcFe) [inline frontend/src/components/activePage/TableInstance/TableInstance.tsx:108]: I'm not extremely sure of this...

File: frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  /* const instances = instanceList.map(i => {
+    const item = i.split('/');
+    return (
+      <li>
+        <b>User: </b>
+        {item[0]} - <b>Instance: </b>
+        {item[1]}
+      </li>
+    );
+  }); */
```
Comment (QcFe) [inline frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx]: Is this still needed?

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx [hunk 8]
```diff
@@ -188,10 +192,20 @@ const TableInstance: FC<ITableInstanceProps> = ({ ...props }) => {
...
           </Button>
-          <ModalAlert
+          <ModalGroupDeletion
+            view={WorkspaceRole.user}
+            persistent={!!instances.filter(i => i.persistent === true).length}
+            selective={false}
+            instanceList={instances.map(i => i.id)}
+            show={showAlert}
+            setShow={setShowAlert}
+            destroy={() => destroyAll()}
```
Comment (QcFe) [inline frontend/src/components/activePage/TableInstance/TableInstance.tsx:207]:
```diff
-            destroy={() => destroyAll()}
+            destroy={destroyAll}
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx [hunk 9]
```diff
@@ -188,10 +192,20 @@ const TableInstance: FC<ITableInstanceProps> = ({ ...props }) => {
...
+          />
+          {/* <ModalAlert
```
Comment (QcFe) [inline frontend/src/components/activePage/TableInstance/TableInstance.tsx]: Remove commented out code if not needed

File: frontend/src/components/activePage/TableTemplate/TableTemplateRow.tsx [hunk 10]
```diff
@@ -125,7 +125,19 @@ const TableTemplateRow: FC<ITableTemplateRowProps> = ({ ...props }) => {
...
+        view={WorkspaceRole.manager}
+        persistent={
+          !!template.instances.filter(i => i.persistent === true).length
+        }
+        groupName={template.name}
+        selective={false}
+        instanceList={template.instances.map(i => i.id)}
+        show={showAlert}
+        setShow={setShowAlert}
+        destroy={() => destroyAll()}
```
Comment (QcFe) [inline frontend/src/components/activePage/TableTemplate/TableTemplateRow.tsx]:
```diff
-        destroy={() => destroyAll()}
+        destroy={destroyAll}
```

File: frontend/src/components/activePage/TableTemplate/TableTemplateRow.tsx [hunk 11]
```diff
@@ -125,7 +125,19 @@ const TableTemplateRow: FC<ITableTemplateRowProps> = ({ ...props }) => {
...
+      />
+      {/* <ModalAlert
```
Comment (QcFe) [inline frontend/src/components/activePage/TableTemplate/TableTemplateRow.tsx]: Commented code :)

File: frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+  const title = selective ? 'Destroy Selected' : 'Destroy All';
+  const message = <b>ATTENTION</b>;
+  const description = (
+    <>
+      <div>
+        Are you sure do you want to destroy
+        {selective ? (
+          <>
+            {` the ${instanceList.length} selected instances`}
+            {/* <ul>{instances}</ul> */}
```
Comment (QcFe) [inline frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx]: Commented out code

File: frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+      }}
+    >
+      {selective ? 'Destroy Selected' : 'Destroy All'}
+    </Button>,
+  ];
+
+  const checkbox = {
+    confirmCheckbox: confirmDeletion,
+    setConfirmCheckbox: setConfirmDeletion,
+    checkboxLabel: 'I undertand the risk and I want to proceed',
```
Comment (QcFe) [inline frontend/src/components/activePage/ModalGroupDeletion/ModalGroupDeletion.tsx]:
```diff
-    checkboxLabel: 'I undertand the risk and I want to proceed',
+    checkboxLabel: 'I understand the risk and I want to proceed',
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx [hunk 14]
```diff
@@ -54,16 +54,20 @@ const TableInstance: FC<ITableInstanceProps> = ({ ...props }) => {
...
+          variables: {
+            instanceId: instance.name,
+            tenantNamespace: instance.tenantNamespace!,
+          },
+        });
       });
-    });
   };
 
+  const disabled = !instances.filter(i => i.persistent === false).length;
```
Comment (QcFe) [inline frontend/src/components/activePage/TableInstance/TableInstance.tsx:69]:
```diff
-  const disabled = !instances.filter(i => i.persistent === false).length;
+  const disabled = !instances.find(i => i.persistent === false);
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx [hunk 15]
```diff
@@ -188,10 +192,20 @@ const TableInstance: FC<ITableInstanceProps> = ({ ...props }) => {
...
               setShowAlert(true);
             }}
+            disabled={disabled}
           >
             Destroy All
           </Button>
-          <ModalAlert
+          <ModalGroupDeletion
+            view={WorkspaceRole.user}
+            persistent={!!instances.filter(i => i.persistent === true).length}
```
Comment (QcFe) [inline frontend/src/components/activePage/TableInstance/TableInstance.tsx:202]:
```diff
-            persistent={!!instances.filter(i => i.persistent === true).length}
+            persistent={!!instances.find(i => i.persistent)}
```

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 16]
```diff
@@ -160,8 +160,12 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
       return i;
     });
 
+  //console.log(dataTemplate);
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx]: This is still here... (also line 167) 🧐

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx (hunk 8 above)
Comment (QcFe) [reply frontend/src/components/activePage/TableInstance/TableInstance.tsx:207]:
```suggestion
            destroy={destroyAll}
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx (hunk 15 above)
Comment (QcFe) [reply frontend/src/components/activePage/TableInstance/TableInstance.tsx:202]:
```suggestion
            persistent={!!instances.find(i => i.persistent)}
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx (hunk 14 above)
Comment (QcFe) [reply frontend/src/components/activePage/TableInstance/TableInstance.tsx:69]:
```suggestion
  const disabled = !instances.find(i => i.persistent === false);
```

File: frontend/src/components/activePage/TableInstance/TableInstance.tsx (hunk 6 above)
Comment (cannarelladev) [reply frontend/src/components/activePage/TableInstance/TableInstance.tsx:108]: It's only for the header, and this is unique in the component

**Review by QcFe — APPROVED** [review]

**Pull request merged**